sign/impl.go

```diff
@@ -102,3 +105,18 @@ func (*defaultImpl) IsImageSignedInternal(
 
 	return len(signatures) > 0, nil
 }
+
+// TokenFromProviders will try the cosign OIDC providers to get an
+// oidc token from them.
+func (*defaultImpl) TokenFromProviders(ctx context.Context) (string, error) {
+	if !providers.Enabled(ctx) {
+		logrus.Warn("No OIDC provider enabled. Token cannot be obtained autmatically.")
+		return "", nil
+	}
```

Comment:
I'd move this into a dedicated method for the `defaultImpl`, because we now mock logical paths rather than single testable units.

Reply:
OK, I've moved it to another function, I'll do a test for these new ones in a follow up :+1: